backend/src/main/java/com/benine/backend/ServerController.java

```diff
@@ -47,13 +44,13 @@ private ServerController(String configPath) {
     running = false;
     setupLogger();
 
-    database = loadDatabase();
-
-    cameraController = new CameraController();
+    databaseController = new DatabaseController(this);
+    
+    streamController = new StreamController();
 
-    presetController = new PresetController();
+    cameraController = new CameraController(this);
```

Comment:
Why not make Servercontroller static? Then you won't have to pass "this" to all the constructors.